addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -271,46 +126,79 @@ var LinkDialog = Dialog.extend({
         }
     },
     /**
-     * @private
+     * @override
+     */
+    _isNewWindow: function (url) {
+        if (this.options.forceNewWindow) {
+            return this._isFromAnotherHostName(url);
+        } else {
+            return this.$('input[name="is_new_window"]').prop('checked');
+        }
+    },
+    /**
+     * @override
      */
-    _correctLink: function (url) {
-        if (url.indexOf('mailto:') === 0 || url.indexOf('tel:') === 0) {
-            url = url.replace(/^tel:([0-9]+)$/, 'tel://$1');
-        } else if (url.indexOf('@') !== -1 && url.indexOf(':') === -1) {
-            url = 'mailto:' + url;
-        } else if (url.indexOf('://') === -1 && url[0] !== '/'
-                    && url[0] !== '#' && url.slice(0, 2) !== '${') {
-            url = 'http://' + url;
+    _setSelectOption: function ($option, active) {
+        if ($option.is("input")) {
+            $option.prop("checked", active);
+        } else if (active) {
+            $option.parent().find('option').removeAttr('selected').removeProp('selected');
+            $option.parent().val($option.val());
+            $option.attr('selected', 'selected').prop('selected', 'selected');
         }
-        return url;
+    },
+    /**
+     * @override
+     */
+    _updateOptionsUI: function () {
+        const el = this.el.querySelector('[name="link_style_color"]:checked');
+        $(this.buttonOptsCollapseEl).collapse(el && el.value ? 'show' : 'hide');
     },
 
     //--------------------------------------------------------------------------
     // Handlers
     //--------------------------------------------------------------------------
 
-    /**
-     * @private
-     */
-    _onAnyChange: function () {
-        this._adaptPreview();
-    },
     /**
      * @private
      */
     _onTypeChange() {
         this._updateOptionsUI();
     },
     /**
-     * @private
+     * @override
      */
     _onURLInput: function () {
-        var $linkUrlInput = this.$('#o_link_dialog_url_input');
-        $linkUrlInput.closest('.form-group').removeClass('o_has_error').find('.form-control, .custom-select').removeClass('is-invalid');
-        let value = $linkUrlInput.val();
-        let isLink = value.indexOf('@') < 0;
-        this.$('input[name="is_new_window"]').closest('.form-group').toggleClass('d-none', !isLink);
-        this.$('.o_strip_domain').toggleClass('d-none', value.indexOf(window.location.origin) !== 0);
+        this._super(...arguments);
+        this.$('#o_link_dialog_url_input').closest('.form-group').removeClass('o_has_error').find('.form-control, .custom-select').removeClass('is-invalid');
+    },
+});
+
+/**
+ * Allows to customize link content and style.
+ */
+ const LinkDialog = Dialog.extend({
+    init: function () {
+        this.linkWidget = new _DialogLinkWidget(...arguments);
+        this._super(...arguments);
+    },
```

Comment:
I would rather do:
```js
    init: function (parent, ...args) {
        this.linkWidget = new _DialogLinkWidget(this, args);
        this._super(...arguments);
    },
```
In the Odoo widget structure, it is intended to have this kind of hierarchy. It is more safe and coherent to keep the common practice.

Reply:
That code crashes.